src/Uno.UI/Microsoft/UI/Xaml/Controls/NavigationView/NavigationViewItem.cs

```diff
@@ -1108,7 +1108,12 @@ private void ProcessPointerCanceled(PointerRoutedEventArgs args)
 			// What this flag tracks is complicated because of the NavigationView sub items and the m_capturedPointers that are being tracked..
 			// We do this check because PointerCaptureLost can sometimes take the place of PointerReleased events.
 			// In these cases we need to test if the pointer is over the item to maintain the proper state.
-			if (IsOutOfControlBounds(args.GetCurrentPoint(this).Position))
+			if (IsOutOfControlBounds(args.GetCurrentPoint(this).Position)
+#if HAS_UNO
+				// UNO#7150 workaround: pointer can be still within when the events are raised
+				|| (args.Pointer.PointerDeviceType == PointerDeviceType.Touch && !args.Pointer.IsInContact)
+#endif
```

Comment:
This change makes sens, I do had to do something similar for pointer over. I'm wondering however why this change is needed only for Uno, the issue should be the same on UWP. Are we able to repro with touch on XCG?

Note: As `Touch` is the common pointer on platform that do have the less power, you should inverse the 2 conditions: the `args.GetCurrentPoint(this)` can be pretty expensive, if we can avoid it, we should!